libopendroneid/opendroneid.c

```diff
@@ -17,44 +17,63 @@ gabriel.c.cox@intel.com
 #include <stdio.h>
 #define ENABLE_DEBUG 1
 
-const float SPEED_DIV[2] = {0.25,0.75};
+const float SPEED_DIV[2] = {0.25f, 0.75f};
 const float VSPEED_DIV = 0.5;
 const int32_t LATLON_MULT = 10000000;
 const float ALT_DIV = 0.5;
 const int ALT_ADDER = 1000;
 const int DATA_AGE_DIV = 10;
 
+/**
+* Encode direction as defined by Open Drone ID
+*
+* The encoding method uses 8 bits for the direction in degrees and
+* one extra bit for indicating the East/West direction.
+*
+* @param Direcction in degrees (typically between 0 and 359)
```

Comment:
Typically, or required?

Reply:
I was wondering that also when I did the implementation. The main problem is that we can't control what people will input. I guess the easiest would be to specify that the value must be 0 <= x < 360. It is a float value so it could get quite close to 360 but should not be equal to or larger. 
I implemented the code so that it actually accepts a larger range but will clamp negative values to zero and do modulus on values larger than 360.